networkx/algorithms/matching.py

```diff
@@ -33,6 +33,12 @@ def maximal_matching(G):
     matching : set
         A maximal matching of the graph.
 
+    Examples
+    --------
+    >>> G=nx.Graph ([(1, 2), (1, 3), (2, 3), (2, 4), (3, 5), (4, 5)])
+    >>> nx.maximal_matching(G)
+    {(1, 2), (3, 5)}
```

Comment:
Checking sets in doctests is not stable as the order of sets is not guaranteed. One solution would be to use `sorted()` to ensure the order is the same every time. See [the doctest docs](https://docs.python.org/3/library/doctest.html#warnings) for more info.

Reply:
Thanks, sir, I have adjusted it.